### ISSUE TITLE
Dynamic finders should be overwritten when a method with that name exists

### ISSUE THREAD

**Issue by dylanpyle** [balderdashy/waterline#926]
Situation: I've defined a model as follows:

```
Items =

  attributes: {
    type: {
      type: 'text'
      required: true
    }
     
    isDisabled: {
      type: 'boolean'
      defaultsTo: false
      required: true
    }
  }

  findByType: function(type) {
    return this.find({
      isDisabled: false,
      type: type
    }) // other stuff e.g. injecting related things
  }
```

Thanks to the magic of dynamic finders, there's already a `findByType` method before I define mine. But in a case where I wasn't aware of all of these or wanted to re-define it anyway (without disabling all the others either globally of for this model), I'd want to be able to override it by default.

**Expected result:**  When I call `Items.findByType('blue')` later on, my custom `findByType` method gets called.

**Actual result:** `Items.findByType` is still the Waterline dynamic finder. No errors or other indication - annoyingly subtle because it *almost* does the same thing as the method I expected.

**Current workarounds:** Renaming this method, or setting `dynamicFinders: false` on this model and avoiding using the other ones.

**Comment by dmarcelino** [comment]
I think this is a reasonable feature and it shouldn't be too complicated to implement.

The dynamic finders code is in: https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/finders/dynamicFinders.js

And tests in: https://github.com/balderdashy/waterline/blob/master/test/unit/query/query.dynamicFinders.js

**Comment by mikermcneil** [comment]
Thanks for posting, @dylanpyle.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/waterline/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
